actions/class.ItemContent.php

```diff
@@ -205,9 +206,15 @@ public function download(): void
         $resolver = new ItemMediaResolver($item, $itemLang);
 
         $asset = $resolver->resolve($params['path']);
-        $stream = $asset->getMediaSource()->getFileStream($asset->getMediaIdentifier());
+        $mediaSource = $asset->getMediaSource();
 
-        $info = $asset->getMediaSource()->getFileInfo($asset->getMediaIdentifier());
+        if ($mediaSource instanceof ProcessedFileStreamAware) {
```

Comment:
Can we extract this at least to separate method?

Reply:
Refactored